### PR TITLE
Fix pypi publish pipeline, Follow-up to #1053

### PR DESCRIPTION
**Pull request by hrshdhgd**
Follow-up to #1053

### CODE REVIEW

**Comment by dalito** [comment]
@hrshdhgd - Maybe you want to pipx-install poetry like in https://github.com/linkml/linkml-runtime/pull/214? Then poetry is isolated from the package. pipx is available by default.

**Comment by hrshdhgd** [comment]
@dalito , I am trying it with `pip install -U poetry` in #1083

**Comment by hrshdhgd** [comment]
Even when I do `poetry install` locally, I get:
```
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: linkml (0.0.0)
```

Is this different for you? It blindly grabs the version from the toml file.

**Comment by dalito** [comment]
@hrshdhgd - Can you add a line in CI code to list the poetry plugins for debugging:
```
poetry self show plugins
```

**Comment by hrshdhgd** [comment]
I get `Too many arguments.` in red ..lol!

**Comment by dalito** [comment]
With poetry-1.2.2?

**Comment by hrshdhgd** [comment]
Ok, problems I noticed:

 - I was using an older version of `poetry` locally
 - Due to the above `poetry update` did not update the lock file with the plugins.

I think we're inching close towards the solution

**Comment by hrshdhgd** [comment]
Nope: I still get:

```
• poetry-plugin-export (1.1.2) Poetry plugin to export the dependencies to various formats
      1 application plugin

      Dependencies
        - poetry (>=1.2.0,<2.0.0)
        - poetry-core (>=1.1.0,<2.0.0)
```

which does not show what you are getting

**Comment by dalito** [comment]
If this fails I next guess is that builing before installing could help:
```
        run: |
          poetry build
          poetry install -E docs
          poetry self show plugins```

**Comment by hrshdhgd** [comment]
I tried this locally, still get what I got before. No change.

**Comment by dalito** [comment]
Hmm. Does the release pipeline re-use something from the main pipeline? Because there the somewhat problematic `snok/install-poetry` is still used.

**Comment by hrshdhgd** [comment]
ok, I didn't do Step 1 explained in the https://github.com/mtkennerly/poetry-dynamic-versioning repo

 - `poetry self add "poetry-dynamic-versioning[plugin]"`

File: .github/workflows/pypi-publish.yaml [hunk 1]
```diff
@@ -12,23 +12,18 @@ jobs:
...
-      #   with:
-      #     virtualenvs-create: true
-      #     virtualenvs-in-project: true
+      - name: Install Poetry
+        run: pip install poetry
 
       - name: Install dependencies
-        run: poetry install --no-interaction
+        run: |
+          poetry self add "poetry-dynamic-versioning[plugin]"
```
Comment (dalito) [inline .github/workflows/pypi-publish.yaml:25]: You could move the installation the pipx / poetry section. The command is: 
```
pipx inject poetry poetry-dynamic-versioning
```

**Comment by hrshdhgd** [comment]
This did the trick! Thank you so much for the help @dalito!

File: .github/workflows/pypi-publish.yaml (hunk 1 above)
Comment (hrshdhgd) [reply .github/workflows/pypi-publish.yaml:25]: Done! Thanks again!

**Review by cmungall — APPROVED** [review]

**Pull request merged**